### DISCUSSION TITLE
Sidekiq dies in docker container

### DISCUSSION THREAD

**Discussion by meap** [sidekiq/sidekiq#5085] [Q&A]
In the case of a task that runs out of CPU or memory, AWS ECS will kill the container. I have set retries of the task to 2 attempts. Once the container is killed, Sidekiq correctly queues the job back so it can be processed again. Until now, it's the right behavior.

On the second attempt when the container is killed (retries are exhausted) Sidekiq dies without triggering e.g. `sidekiq_retries_exhausted`.

What should be the right behaviour? Is this a bug or am I just not understanding something?

**Comment by mperham** [comment]
What version(s) of Sidekiq? Are you using super_fetch?

**Reply by meap** [reply]
```
    sidekiq (6.3.1)
      connection_pool (>= 2.2.2)
      rack (~> 2.0)
      redis (>= 4.2.0)
    sidekiq-pro (5.3.0)
      sidekiq (>= 6.3.0)
```

@mperham yes, I use super_fetch.

**Reply by mperham** [reply]
It sounds like ECS is killing the process without warning. This makes it impossible for Sidekiq to do anything, including calling `sidekiq_retries_exhausted`.

Once a job is recovered three times by super_fetch, it goes to the Dead queue. I suspect you'll find your jobs on the Dead tab.

**Reply by meap** [reply]
Correct, I wonder why the mentioned event is not raised when inserted into the dead queue?

**Reply by meap** [reply]
Or is there another way to handle a situation where super_fetch puts such a job in the dead queue?

**Reply by mperham** [reply]
A retry is a response to a job raising an exception. In this case, there's no exception to handle: the process just disappears. super_fetch recovers the job data from the bowels of Redis but this codepath is not a "retry". Your job is better classified as a poison pill: https://github.com/mperham/sidekiq/issues/4633
